lib/net.js

```diff
@@ -906,8 +906,11 @@ Socket.prototype.connect = function(options, cb) {
     if (localPort && !util.isNumber(localPort))
       throw new TypeError('localPort should be a number: ' + localPort);
 
-    if (port <= 0 || port > 65535)
-      throw new RangeError('port should be > 0 and < 65536: ' + port);
+    if (options.port !== undefined && parseInt(options.port, 10) !== port)
```

Comment:
Right, I thought `"42foo" | 0` would evaluate to `42`, which it does not. Thanks for the clarification!

Using `parseInt` only with a radix of `10` makes strings representing hexadecimal numbers such as `0x12` throw. The current version of the code does, and it seems that it matches with the concept of "undefined, a number, or a string representing a number".

I'm not sure if we want to support it though. I added support for strings representing hexadecimal numbers [in a gist](https://gist.github.com/misterdjules/7e4f25997b3805792aac), and the problem is that it would accept invalid values such as `0x123R`, since `0x123R | 0 === 0` and `parseInt(0x123R, 10) === 0` too.

Other than that, I think the above-mentioned gist improves the current changes by making the checks a bit clearer though, so feel free to reuse it without the hexadecimal string support.